### PR TITLE
Bugfix; do not allocate more than needed

### PR DESCRIPTION
**Pull request by PallHaraldsson**

### CODE REVIEW

**Comment by PallHaraldsson** [comment]
This may also fix a bug. Empty string now allocates 2 bytes, and it may be needed for a terminator.

File: cli/loader_win_utils.c [hunk 1]
```diff
@@ -15,8 +15,8 @@ FILE *stderr = &_stderr;
 int loader_fwrite(const char *str, size_t nchars, FILE *out) {
     DWORD written;
     if (out->isconsole) {
-        // Windows consoles do not support UTF8, only UTF16.
-        size_t wbufsz = nchars * 4 + 1;
+        // Windows consoles do not support UTF-8 (for reading input, though new Windows Terminal does for writing), only UTF-16.
+        size_t wbufsz = ((nchars * 3) >> 1) + 2;
```
Comment (vtjnash) [inline cli/loader_win_utils.c:19]:
```suggestion
        size_t wbufsz = nchars * 2 + 2;
```

Comment (PallHaraldsson) [reply cli/loader_win_utils.c:19]: Do you really think it needs 2x not 50%? When?

Comment (vtjnash) [reply cli/loader_win_utils.c:19]: Code points U+0000 to U+007F (ASCII space) require 1 utf8 byte and 2 utf16 bytes. It is the most common case (for filenames).

**Comment by PallHaraldsson** [comment]
You mean line 20(?!) or more likely actually replace line 21 with it? I only want to fix what I understand.

File: cli/loader_win_utils.c (hunk 1 above)
Comment (PallHaraldsson) [reply cli/loader_win_utils.c:19]: Isn't it always good to allocate least possible? It's only one more assembly instruction.

**Comment by PallHaraldsson** [comment]
Good to just merge? A simple change even though I confused myself...

**Comment by vtjnash** [comment]
Sorry, I had line 25 in mind (the WriteConsole call) actually

**Review by vtjnash — APPROVED** [review]

**Comment by PallHaraldsson** [comment]
Can someone other than me squash commits? I'm busy and don't recall how (also using web UI, so if I recall not possible). I just dislike the wrong "50%" more in the first commit...

**Comment by gbaraldi** [comment]
The commits get squashed when merged, it's fine :)

**Pull request merged**